mono/metadata/w32file-unix.c

```diff
@@ -3578,26 +3579,33 @@ mono_w32file_get_attributes_ex (const gunichar2 *name, MonoIOStat *stat)
 	stat->attributes = _wapi_stat_to_file_attributes (utf8_name, &buf, &linkbuf);
 	stat->length = (stat->attributes & FILE_ATTRIBUTE_DIRECTORY) ? 0 : buf.st_size;
 
+	if (&linkbuf == NULL) {
+		pbuf = &buf;
```

Comment:
`&linkbuf` will _never_ be `NULL` as it's allocated on the stack so it always has an address. In the end, your change boils down to `pbuf = &linkbuf` all the time. And this is fine, as `lstat` behaves the same as `stat` if the provided file isn't a symbolic link.